lib/class-oa2-authenticator.php

```diff
@@ -34,14 +34,13 @@ public function __construct() {
    * @return WP_User|null|WP_Error Authenticated user on success, null if no OAuth data supplied, error otherwise
    */
   public function authenticate( $user ) {
-	if ( !empty( $user ) ) {
-	  return $user;
-	}
-
 	// Check if token is in request
 	$bearer_token = OA2_Header_Helper::get_authorization_bearer();
 	if ( empty( $bearer_token ) ) {
 	  $this->auth_status = $bearer_token;
+      if ( !empty( $user ) ) {
```

Comment:
Could you elaborate what was the problem that this change fixed? The logic follows the official OAuth1 plugin (https://github.com/WP-API/OAuth1/blob/master/lib/class-wp-rest-oauth1.php#L166) as the already authenticated user is supposed to take precendence, if it exists.